setup.py

```diff
@@ -552,6 +552,9 @@ def mac_osx_setenv():
     logging.info("Setting SDKROOT=%s", sdk_root)
     os.environ["SDKROOT"] = sdk_root
 
+    # Clear Python arch flags which can be incompat with the cur system
```

Comment:
What do you mean by `Python arch flags`?  What is setting `ARCHFLAGS`?
Also, doesn't this impact the creation of universal2 binaries, or cross-compiling for `arm64` on `x86_64` and vica-versa?

Reply:
You can check for example `sysconfig.get_config_var("PY_CORE_CFLAGS")` that we get the `-arch` flag twice with different architectures. I'm not entirely sure why they would want to simultaneously produce code for two architectures. That is likely hard to support support and I can't imagine we require it since we don't cross compile. Also strangely that was not the case for previous Pythons 🤷‍♂️
Setting to empty seems to allow auto-detection by the compiler which is likely what we want, but all this is very experimental, it might be that the final release of Python doesn't require this.

For the moment this is the only way I know to be able to build Neuron with Py3.13 in Mac

Reply:
Isn't `PY_CORE_CFLAGS` from how python was built, or what the installed version of xcode supports; so if it was cross compiled, that's why you get two `-arch`s?

>  Also strangely that was not the case for previous Pythons

Do you think it's a change to setuptools?  Which versions are you using that are failing?

> Setting to empty seems to allow auto-detection by the compiler which is likely what we want

I don't think it's an auto-detection, I think it just uses the default.
Having cross compilation available is very handy when someone wants to test whether a particular build works by passing it to someone who doesn't necessarily have all the infrastructure to build it themselves.

> For the moment this is the only way I know to be able to build Neuron with Py3.13 in Mac

What else have you tried?  Maybe include what version of xcode you are running, and other configuration that is tripping up the change.

Reply:
>  Also strangely that was not the case for previous Pythons

I saw someone with the same issue with another package on an earlier version of MacOS/Python. Seemed to indicate that it's an issue with Python's official Mac builds and that `brew`'s Python did not have that issue.

Reply:
I'd maybe put a note somewhere that `export ARCHDIR=....` is required before `python setup.py ...`

Reply:
> Isn't PY_CORE_CFLAGS from how python was built, or what the installed version of xcode supports; so if it was cross compiled, that's why you get two -archs?

yep `PY[_CORE]_CFLAGS` is how it was built, and definitely the toolchain was different than mine. I am now more convinced they use newer mac SDKs to build for both intel and apple silicon.

While it's truly a good habit to use the same flags (hence setuptools picks them, since I know setuptools), in this case it wouldn't work, likely the same story for everyone with an older SDK.

> I'd maybe put a note somewhere that export ARCHDIR=.... is required before python setup.py ...

I agree, in the end this is information helpful for some specific build conditions, not a general rule. I'll change this to a note somewhere in the "developer builds" section of the docs